Assign rooms reworked

movingpylon opened the pull request:
Note: dependents list in assign rooms now only shows users approved for the conference.

nikkimartin commented:
- Dependent names don't show up in registration details.
- require accommodations doesn't show up in registration details
- at assignment step, right panel is saying no rooms available and has the enter a room name and assign? (try assigning a room from barbaz res, should show you what I mean)

movingpylon commented:
Registration details page is now fixed.

As for room assignments, I think you are supposed to be able to create a new room and assign a user to it, regardless of whether or not rooms already exists. I changed the labels for this to make it less confusing.
